core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SequencingParameters.java

```diff
@@ -143,4 +144,32 @@ public void setUpdatedBy(User updatedBy) {
     this.updatedBy = updatedBy;
   }
 
+  public static Comparator<SequencingParameters> nameComparator = (sp1, sp2) -> sp1.compareTo(sp2);
+
+  @Override
+  public int compareTo(SequencingParameters other) {
+    // v4 2x126, v4, Rapid Run, 10X, Custom, v3, other
+    if (!(other instanceof SequencingParameters)) return -1;
+    if (name.startsWith("v4")) {
+      if ("v4 2×126".equals(name)) return -1;
+      if ("v4 2×126".equals(other.name)) return 1;
+      if (other.name.startsWith("v4")) return other.name.compareTo(name);
+      return -1; // v4 > everything else
+    } else if (name.startsWith("Rapid Run")) {
+      if (other.name.startsWith("v4")) return 1;
+      if (other.name.startsWith("Rapid Run")) return other.name.compareTo(name);
+      return -1; // Rapid Run > everything else
+    } else if (name.startsWith("10X")) {
+      if (other.name.startsWith("v4") || other.name.startsWith("Rapid Run")) return 1;
+      if (other.name.startsWith("10X")) return other.name.compareTo(name);
+      return -1; // 10X > everything else
+    } else if (name.startsWith("v3")) {
+      if (other.name.startsWith("v4") || other.name.startsWith("Rapid Run") || other.name.startsWith("10X")) return 1;
```

Comment:
A more manageable way to write this would be to assign an int to each pattern (1 for `v4 2x126`, 2 for `v4*`, etc.), compare the ints first, and then compare names if the ints are equal. Would result in much less conditional logic, especially if/when new patterns are added